### PR TITLE
Add new formula goolabs 0.1.1

### PR DESCRIPTION
**Pull request by tell-k**
Can morphological analysis of Japanese, it is a command-line tool.
Using the WebAPI of goolab(https://labs.goo.ne.jp/api/).

### CODE REVIEW

File: Library/Formula/goolabs.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require "formula"
```
Comment (bfontaine) [inline Library/Formula/goolabs.rb]: The test is failing because this line is not necessary anymore.

Comment (tell-k) [inline Library/Formula/goolabs.rb]: @bfontaine thx.
I just removed「require "formula"」.

File: Library/Formula/goolabs.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    You may pass the App id every time you use it, but it's not recommended.
+
+      goolabs morph --app-id [your goolabs app id] 日本語を分析します。
+
+    EOS
+  end
+
+  test do
+    system bin+"goolabs", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/goolabs.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/goolabs.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    To use this cli, it is recommended to set the environment variable GOOLABS_APP_ID.
+
+      # Write your shell setting files(ex ~/.bashrc).
+      export GOOLABS_APP_ID=[your goolabs app id]
+
+    You may pass the App id every time you use it, but it's not recommended.
+
+      goolabs morph --app-id [your goolabs app id] 日本語を分析します。
+
+    EOS
```
Comment (mikemcquaid) [inline Library/Formula/goolabs.rb]: Unless these caveats are Homebrew-specific just remove them, thanks!

File: Library/Formula/goolabs.rb (hunk 2 above)
Comment (tell-k) [inline Library/Formula/goolabs.rb]: @mikemcquaid  Sorry my late reply.
I just improved test. Please feel to merge. thx.

File: Library/Formula/goolabs.rb (hunk 3 above)
Comment (tell-k) [inline Library/Formula/goolabs.rb]: @mikemcquaid   I just removed caveats. Please check it again. Thx.

File: Library/Formula/goolabs.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    require "open3"
+    Open3.popen3("#{bin}/goolabs", "morph", "test") do |stdin, stdout, stderr, _|
+        assert_equal "Usage: goolabs morph [OPTIONS] [SENTENCE]\n\nError: Missing option \"--app-id\" / \"-a\" or GOOLABS_APP_ID enviroment value.\n", stderr.read
+    end
```
Comment (mikemcquaid) [inline Library/Formula/goolabs.rb]: Just do `assert_match "Usage: goolabs morph", system_output "goolabs morph test 2>&1"`

Comment (tell-k) [inline Library/Formula/goolabs.rb]: I completed it. Please check it again. thx.

File: Library/Formula/goolabs.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (DomT4) [inline Library/Formula/goolabs.rb]: Can squash the empty line up top here.

File: Library/Formula/goolabs.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    end
+
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+     require "open3"
```
Comment (DomT4) [inline Library/Formula/goolabs.rb]: Indentation looks off here. Should be two spaces; I think this is three, or maybe four.

File: Library/Formula/goolabs.rb (hunk 5 above)
Comment (tell-k) [inline Library/Formula/goolabs.rb]: @DomT4 I fixed it. thx

File: Library/Formula/goolabs.rb (hunk 6 above)
Comment (tell-k) [inline Library/Formula/goolabs.rb]: @DomT4 I fixed it. Please feel free to merge. thx

File: Library/Formula/goolabs.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    require "open3"
+    Open3.popen3("#{bin}/goolabs morph test 2>&1") do |stdin, stdout, _|
+       assert_match "Usage: goolabs morph", stdout.read
+    end
```
Comment (mikemcquaid) [inline Library/Formula/goolabs.rb]: `assert_match "Usage: goolabs morph", system_output("#{bin}/goolabs morph test 2>&1")`

Comment (tell-k) [inline Library/Formula/goolabs.rb]: @mikemcquald  I can't use "system_output". Is this the right way?
```
$ brew test goolabs 
Testing goolabs
Error: goolabs: failed
undefined method `system_output' for #<Formula goolabs (stable) /usr/local/Library/Formula/goolabs.rb>
/usr/local/Library/Formula/goolabs.rb:42:in `block in <class:Goolabs>'
/usr/local/Library/Homebrew/formula.rb:639:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:21:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:636:in `run_test'
/usr/local/Library/Homebrew/cmd/test.rb:38:in `block (2 levels) in test'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `block in test'
/usr/local/Library/Homebrew/cmd/test.rb:15:in `each'
/usr/local/Library/Homebrew/cmd/test.rb:15:in `test'
/usr/local/Library/brew.rb:135:in `<main>'
```

Comment (mikemcquaid) [inline Library/Formula/goolabs.rb]: Sorry, `shell_output`, my mistake.

Comment (tell-k) [inline Library/Formula/goolabs.rb]: @mikemcquaid  Thank you for your help.  I fixed it using "shell_output".

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `goolabs 0.1.1 (new formula)`.

**Comment by tell-k** [comment]
@mikemcquaid  Thank you for telling me many things! You rock too!

**Pull request closed without merging**